frontend/src/components/common/Address.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import * as Strings from "@/constants/strings";
+import { Icon } from "antd";
+import { formatPostalCode } from "@/utils/helpers";
+
+const propTypes = {
+  address: PropTypes.objectOf(PropTypes.string).isRequired,
+};
+
+/**
+ * @constant Address- reusable component that formats any given address object
+ */
+
+export class Address extends Component {
+  addressContainsTruthyValues = (object) =>
+    // address_type_code is always "CAD" && if "suit_no" exists but nothing else, this field isn't helpful
+    Object.keys(object)
+      .filter((key) => key !== "address_type_code" && key !== "suite_no")
+      .some((key) => object[key]);
+
+  formatRowContent = (itemArr) =>
+    itemArr
+      .filter((item) => item)
+      .map((item, i) => (
+        <span key={item}>{i !== itemArr.length - 1 ? `${item}${" "}` : item}</span>
+      ));
+
+  renderRowOne = (itemArr) => (
+    <div className="inline-flex">
+      <div>
+        <Icon type="contacts" className="icon-sm" />
+      </div>
+      <p>{this.formatRowContent(itemArr)}</p>
+    </div>
+  );
+
+  renderNextLine = (itemArr) => (
+    <div className="padding-large--left">
+      <p>{this.formatRowContent(itemArr)}</p>
+    </div>
+  );
+
+  // since nothing in the address object is required, users can fill out 1 or more fields.
+  // the following check tests all the edge cases, and formats the address into a 1-3 line section, with an Icon on line 1.
+  renderAddress = (address) => (
```

Comment:
If you have time, this would be a method to add regression tests for if possible. It's a pure function, so you can easily cover all of the edge-cases based on just a mock `address` object and then compare the actual output with the expected output.

It could also help the next person who touches this component considerably

Reply:
Great suggestion! will work on that this afternoon.

Reply:
Marking this as resolved, since we've discussed that you may do this separately